base/cvd/cuttlefish/host/commands/cvd/selector/instance_group_record.cpp

```diff
@@ -154,16 +123,60 @@ Json::Value LocalInstanceGroup::Serialize() const {
 }
 
 Json::Value LocalInstanceGroup::Serialize(
-    const std::unique_ptr<LocalInstance>& instance) const {
+    const LocalInstance& instance) const {
   Json::Value instance_json;
-  if (!instance) {
-    return instance_json;
-  }
-  instance_json[LocalInstance::kJsonInstanceName] = instance->PerInstanceName();
+  instance_json[LocalInstance::kJsonInstanceName] = instance.PerInstanceName();
   instance_json[LocalInstance::kJsonInstanceId] =
-      std::to_string(instance->InstanceId());
+      std::to_string(instance.InstanceId());
   return instance_json;
 }
 
+Result<LocalInstanceGroup> LocalInstanceGroup::Deserialize(const Json::Value& group_json) {
+  const std::string group_name = group_json[kJsonGroupName].asString();
+  const std::string home_dir = group_json[kJsonHomeDir].asString();
+  const std::string host_artifacts_path =
+      group_json[kJsonHostArtifactPath].asString();
+  const std::string product_out_path =
+      group_json[kJsonProductOutPath].asString();
+  TimeStamp start_time = CvdServerClock::now();
+
+  // test if the field is available as the field has been added
+  // recently as of b/315855286
+  if (group_json.isMember(kJsonStartTime)) {
+    auto restored_start_time_result =
+        DeserializeTimePoint(group_json[kJsonStartTime]);
+    if (restored_start_time_result.ok()) {
+      start_time = std::move(*restored_start_time_result);
+    } else {
+      LOG(ERROR) << "Start time restoration from json failed, so we use "
+                 << " the current system time. Reasons: "
+                 << restored_start_time_result.error().FormatForEnv();
+    }
+  }
+
+  LocalInstanceGroup group({.group_name = group_name,
+                            .home_dir = home_dir,
+                            .host_artifacts_path = host_artifacts_path,
+                            .product_out_path = product_out_path,
+                            .start_time = std::move(start_time)});
+  const Json::Value& instances_json_array = group_json[kJsonInstances];
+  for (int i = 0; i < instances_json_array.size(); i++) {
+    const Json::Value& instance_json = instances_json_array[i];
+    const std::string instance_name =
+        instance_json[LocalInstance::kJsonInstanceName].asString();
+    const std::string instance_id =
+        instance_json[LocalInstance::kJsonInstanceId].asString();
```

Comment:
Should this use `asInt()` instead to avoid the `ParseInt` check?

Maybe it should instead validate `instance_json[LocalInstance::kJsonInstanceId].type()` first.

Reply:
The field is a string in the file and in the json object passed from one server to the next. Calling asInt() would fail?

Reply:
...but if we assume it is always an int, can it be an int in the json object representation?

Reply:
That object representation needs to match (for now) what the servers from previous versions persist, which are strings.
Once I make it not a server anymore I can maintain that representation when taking over an old server while using whatever representation for the database file.